### PR TITLE
Verx eventbus

### PR DESCRIPTION
**Pull request by aleqi200**

### CODE REVIEW

File: vertx/src/main/java/com/groupon/SumVerticle.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        Observable<Message<List<Integer>>> sumMemory = vertx.eventBus().<List<Integer>>localConsumer(Method.METHOD_EVENTBUS_NETWORK).toObservable();
+        sumMemory.subscribe(message -> callSumNetwork(message.body())
+                .subscribe(body -> {
+                    int sum = Integer.valueOf(body.substring(0, body.indexOf('\n')));
+                    // logger.debug("Received from network: " + sum);
+                    message.reply(sum);
+                }, fail -> message.fail(500, fail.getMessage())));
+    }
+
+    public Observable<String> callSumNetwork(List<Integer> values) {
```
Comment (myok12) [inline vertx/src/main/java/com/groupon/SumVerticle.java]: This is also used as a util method from MainVerticle. Maybe move it to a different class (directly invoking between verticles is weird)?

File: vertx/src/main/java/com/groupon/MainVerticle.java [hunk 2]
```diff
@@ -1,72 +1,56 @@
...
-                        .setTryUseCompression(false));
+        vertx.eventBus().getDelegate().registerDefaultCodec(Collection.class, new CollectionMessageCodec<>());
+        sumVerticle = new SumVerticle(vertx);
+        RxHelper.deployVerticle(vertx, sumVerticle);
         Method[] methods = new Method[]{
                 Method.fromMemory(),
                 new Method(Method.METHOD_NAME_NETWORK, this::sumOverNetwork),
+                new Method(Method.METHOD_EVENTBUS_NETWORK, integers -> {
+                    Single<Message<Integer>> messageSingle = vertx.eventBus().rxSend(Method.METHOD_EVENTBUS_NETWORK, integers, new DeliveryOptions().setCodecName(CollectionMessageCodec.NAME));
+                    return messageSingle.flatMap(event -> Single.just(event.body()));
```
Comment (myok12) [inline vertx/src/main/java/com/groupon/MainVerticle.java]: You should replace with .map() and skip the Single wrapping

Comment (aleqi200) [reply vertx/src/main/java/com/groupon/MainVerticle.java]: oh you're right, removed the single wrap

File: vertx/src/main/java/com/groupon/SumVerticle.java (hunk 1 above)
Comment (aleqi200) [reply vertx/src/main/java/com/groupon/SumVerticle.java]: removed the verticle and turned it into a utility class instead

File: vertx/src/main/java/com/groupon/SumNetworkService.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                        .setMaxWaitQueueSize(-1) // -1 default
+                        //.setPipelining(true)
+                        //.setPipeliningLimit(100)
+                        .setProtocolVersion(HttpVersion.HTTP_1_0) // 1.1 default
+                        .setConnectTimeout(1_000_000)
+                        .setIdleTimeout(1_000) // For some reasons now connections are established
+                        // with 0.
+                        .setTcpNoDelay(true) // Same as in jetty.
+                        .setTryUseCompression(false));
+        Observable<Message<List<Integer>>> sumMemory = vertx.eventBus().<List<Integer>>localConsumer(Method.METHOD_EVENTBUS_NETWORK).toObservable();
```
Comment (myok12) [inline vertx/src/main/java/com/groupon/SumNetworkService.java:37]: Why not just move this into MainVerticle?

**Pull request merged**